src/directory/poll_watcher.rs

```diff
@@ -0,0 +1,73 @@
+use crate::core::META_FILEPATH;
+use crate::directory::error::OpenDirectoryError;
+use crate::directory::{WatchCallback, WatchCallbackList, WatchHandle};
+use std::io;
+use std::path::Path;
+use std::sync::Arc;
+use std::thread;
+use std::time::{Duration, UNIX_EPOCH};
+use std::fs;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+
+const POLL_INTERVAL_MS: u64 = 10;
+
+pub struct PollWatcher {
+    watcher_router: Arc<WatchCallbackList>,
+}
+
+#[derive(Hash)]
+struct State {
+    data: String,
+}
+
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
+
+impl PollWatcher {
+    pub fn new(path: &Path) -> Result<Self, OpenDirectoryError> {
+        let watcher_router: Arc<WatchCallbackList> = Default::default();
+        let watcher_router_clone = watcher_router.clone();
+        let meta_path = path.to_owned().join(*META_FILEPATH);
+        thread::Builder::new()
+            .name("meta-file-watch-thread".to_string())
+            .spawn(move || {
+                let mut current_hash: u64 = Self::hash(&meta_path).unwrap_or(0);
```

Comment:
what is the point of this hash?

Reply:
The problem is: Linux has a 1s granularity for the modified time on files. This means that if 10 updates happen in a second only the first one is caught. There are two possible solutions to this that I can see:
1. Use a hash of the file (which is what I was experimenting with).
2. Delay the update event util the next second to make sure that any update with the same modification time is in the refresh.

Reply:
But you are not hashing the file here. You are hashing the Path. This is very differnt and useless.

About the "mtime" granularity. I think this is 1s is ok. Note also that provided the index is refreshed at interval >1s, the mtime granualrity does not have an impact on the latency of detection (that one is still dependend on `POLL_INTERVAL_MS`).

I am developing a near real time feature on a different branch. People who need an index refreshed more than once per second will simply have to use that.


The check on the hash **on the content of the file** is easy to read, and not very complex, so please keep it. There might be some strange corner cases in which it removes a needless index reload.

Reply:
I have prototypes of SHA256 and MD5 as well. Just wanted to see if it could be implemented without adding any dependencies. If it runs every 500ms the performance should not be that different anyways. 

The reason why I'm testing hashchange or time is that I'm having a lot of strange bugs when running the code on different platforms. Tests pass almost every time on MacOS and inside docker on my computer. In the CI they don't pass. 

Will experiment more in my playground and benchmark the different solutions.